### ISSUE TITLE
Build failure with python 3.5

### ISSUE THREAD

**Issue by opoplawski** [h5py/h5py#644]
We're updating Fedora rawhide to python3.5 and we're hitting a build failure:
~~~~~
creating build/temp.linux-x86_64-3.5/builddir/build/BUILD/python3-h5py-2.5.0-5.fc24/h5py
gcc -pthread -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -fopenmp -fPIC -DH5_USE_16_API -I/builddir/build/BUILD/python3-h5py-2.5.0-5.fc24/lzf -I/opt/local/include -I/usr/local/include -I/usr/lib64/python3.5/site-packages/numpy/core/include -I/usr/include/python3.5m -c /builddir/build/BUILD/python3-h5py-2.5.0-5.fc24/h5py/defs.c -o build/temp.linux-x86_64-3.5/builddir/build/BUILD/python3-h5py-2.5.0-5.fc24/h5py/defs.o
In file included from /usr/include/python3.5m/pyatomic.h:12:0,
                 from /usr/include/python3.5m/Python.h:53,
                 from /builddir/build/BUILD/python3-h5py-2.5.0-5.fc24/h5py/defs.c:36:
/usr/lib/gcc/x86_64-redhat-linux/5.2.1/include/stdatomic.h:40:1: sorry, unimplemented: '_Atomic' with OpenMP
 typedef _Atomic _Bool atomic_bool;
 ^
/usr/lib/gcc/x86_64-redhat-linux/5.2.1/include/stdatomic.h:41:1: sorry, unimplemented: '_Atomic' with OpenMP
 typedef _Atomic char atomic_char;
 ^
~~~~~
Full log: https://kojipkgs.fedoraproject.org//work/tasks/420/11780420/build.log

**Comment by opoplawski** [comment]
Looks like https://bugs.python.org/issue25150 so should be fixed soon.

**Comment by andreabedini** [comment]
Fixed upstream as of Python 3.5.1. Thanks @opoplawski
